### PR TITLE
Cs/refactor msg dialog

### PR DESCRIPTION
**Pull request by changanxian**
- Refactoring the common AlertDialog into a common usage.

### CODE REVIEW

File: lib/samples/show_service_area/show_service_area.dart [hunk 1]
```diff
@@ -260,20 +253,7 @@ class _ShowServiceAreaState extends State<ShowServiceArea>
...
-          ),
-          actions: [
-            TextButton(
-              onPressed: () => Navigator.pop(context, 'OK'),
-              child: const Text('OK'),
-            ),
-          ],
-        ),
-      );
+      showMessageDialog('At least 1 facility is required to perform a service area calculation.', showOK: true,);
```
Comment (PaulAllanSturm) [inline lib/samples/show_service_area/show_service_area.dart:256]: run `dart format .` to get the desired formatting (here and probably elsewhere too)

File: lib/common/dialogs.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//
+
+import 'package:flutter/material.dart';
+
+/// 
+/// Shows an alert dialog with a message.
+/// - [context]: The context in which the dialog is shown.
+/// - [message]: The message to display in the dialog.
+/// - [title]: (Optional) The title of the dialog.
+/// - [showOK]: (Optional) A boolean value to determine if the dialog should show an OK button.
```
Comment (PaulAllanSturm) [inline lib/common/dialogs.dart:24]: The parameter name is "displayOkButton", not "showOK". Probably it should be "showOK" for consistency with sample_state_support.dart.

File: lib/common/dialogs.dart [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+Future<void> showAlertDialog(
+  BuildContext context,
+  String message, {
+  String title = 'Alert',
+  bool displayOkButton = false,
+}) {
+  // if the context State object is in the widget tree.
```
Comment (PaulAllanSturm) [inline lib/common/dialogs.dart:32]: I don't understand this comment.

File: lib/common/dialogs.dart (hunk 2 above)
Comment (changanxian) [reply lib/common/dialogs.dart:24]: updated

**Review by PaulAllanSturm — APPROVED** [review]

**Pull request merged**